include/libdnf-cli/output/search.hpp

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright Contributors to the libdnf project.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2.1 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef LIBDNF_CLI_OUTPUT_SEARCH_HPP
+#define LIBDNF_CLI_OUTPUT_SEARCH_HPP
+
+#include <libdnf/rpm/package.hpp>
+
+#include <set>
+#include <utility>
+#include <vector>
+
+namespace libdnf::cli::output {
+
+/// Type of the match against the related key.
+enum class SearchKeyMatch {
+    PARTIAL,  ///< When the pattern is contained within the given key.
+    EXACT     ///< When the pattern exactly matches the given key.
+};
+
+/// Defines a type for a key that was matched together with a type of the match.
+using matched_key_pair = std::pair<std::string, SearchKeyMatch>;
+
+/// Comparator used for sorting packages within the results.
+static bool search_package_comparator(const libdnf::rpm::Package & one, const libdnf::rpm::Package & two) {
+    return one.get_full_nevra() < two.get_full_nevra();
+};
```

Comment:
This compares the nevra strings, so the packages are not necessarily sorted by version. The result can be e.g.:
```
package-0:1.10-1.noarch
package-0:1.8-1.noarch
package-0:1.9-1.noarch
```
In `include/libdnf/rpm/nevra.hpp`, there are comparators for `Nevra`, but I haven't found a way to get `Nevra` from `Package`. But e.g. `PackageQuery` also implements its own `latest_cmp` for comparing packages by nevra, however, it's not public, so I am wondering if there shouldn't be a shared way to properly compare packages...

Reply:
Nice, thanks for the reference. I can use that comparer directly, because it's templated and the same `Nevra` methods are available also in the `Package` class.